phpunit/class-wp-rest-pattern-directory-controller-test.php

```diff
@@ -0,0 +1,365 @@
+<?php
+/**
+ * Unit tests covering WP_REST_Pattern_Directory_Controller functionality.
+ *
+ * @package WordPress
+ * @subpackage REST API
+ */
+
+/**
+ * @group restapi
+ * @group pattern-directory
+ */
+class WP_REST_Pattern_Directory_Controller_Test extends WP_Test_REST_Controller_Testcase {
+	/**
+	 * Administrator user id.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @var int
+	 */
+	protected static $contributor_id;
+
+	/**
+	 * Set up class test fixtures.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param WP_UnitTest_Factory $factory WordPress unit test factory.
+	 */
+	public static function wpSetUpBeforeClass( $factory ) {
+		self::$contributor_id = $factory->user->create(
+			array(
+				'role' => 'contributor',
+			)
+		);
+	}
+
+	public static function wpTearDownAfterClass() {
+		self::delete_user( self::$contributor_id );
+	}
+
```

Comment:
```suggestion
```

This isn't necessary. It'll be dropped in the base test class with a call to `_delete_all_data`.